chessington/engine/pieces.py

```diff
@@ -33,56 +33,159 @@ class Pawn(Piece):
+            if self.in_bounds(square_two_spaces_in_front.row) and not board.get_piece(square_two_spaces_in_front):
+                list_moves.append(square_two_spaces_in_front)
+
+    def verify_attack_different_color(self, piece):
+        return self.player != piece.player
+
+    def attack_diagonally(self, square_diagonal, board, list_moves):
+        if self.in_bounds(square_diagonal.col):
+            diagonal_piece_left = board.get_piece(square_diagonal)
+            if diagonal_piece_left and self.verify_attack_different_color(diagonal_piece_left):
+                list_moves.append(square_diagonal)
+
+    def verify_attack_diagonally(self, current_square, move_forward, board, list_moves):
+        # Attack diagonally (2 spaces possible)
+        square_diagonal_left = Square.at(current_square.row + move_forward, current_square.col + self.LEFT)
+        square_diagonal_right = Square.at(current_square.row + move_forward, current_square.col + self.RIGHT)
+
+        # Can't attack same color and if a piece isn't there
+        self.attack_diagonally(square_diagonal_left, board, list_moves)
+        self.attack_diagonally(square_diagonal_right, board, list_moves)
+
     def get_available_moves(self, board) -> List[Square]:
         current_square = board.find_piece(self)
-        if self.player == Player.BLACK:
-            square_in_front = Square.at(current_square.row - 1, current_square.col)
-            return [square_in_front]
+        list_moves = []
+        move_forward, row_first_time_move = self.get_moves_by_player_color()
+
+        square_in_front = Square.at(current_square.row + move_forward, current_square.col)
+        # Can move forward at least once
+        if self.in_bounds(square_in_front.row):
+            if not board.get_piece(square_in_front):
+                list_moves.append(square_in_front)
+                # Verify first time move of pawn
+                self.first_time_move(current_square, row_first_time_move, move_forward, board, list_moves)
+            # Verify attack diagonally
+            self.verify_attack_diagonally(current_square, move_forward, board, list_moves)
+
+        return list_moves
+
+
+FIRST_STEP = 1
+
+
+def in_bounds(move) -> bool:
+    return move.row in range(0, 8) and move.col in range(0, 8)
+
+
+def get_position(current_square, current_piece, board, list_moves, step, direction):
+    next_square = Square.at(current_square.row + direction.row * step,
+                            current_square.col + direction.col * step)
+    if in_bounds(next_square):
+        piece = board.get_piece(next_square)
+        if not piece:
+            list_moves.append(next_square)
         else:
-            square_in_front = Square.at(current_square.row + 1, current_square.col)
-            return [square_in_front]
+            if piece.player != current_piece.player:
+                list_moves.append(next_square)
+            return 0
+    else:
+        return 0
+    return 1
```

Comment:
What do these mean? Wouldn't it be clearer to return some constants? `0` and `1` could mean a lot of things